### ISSUE TITLE
About the transfer loss in pytorch implementation

### ISSUE THREAD

**Issue by daiquanyu** [thuml/CDAN#2]
Hi, I'm confused about how you implement the entropy conditioning loss as presented in Equation (9) of the paper. I wonder whether you could describe the tricks you used in the implementation, since it is inconsistent with the paper. Specifically, in Line 34 below, I don't understand why 1 is added to the entropy.

![2018-12-17 21-11-56](https://user-images.githubusercontent.com/29226171/50089397-b4715700-0240-11e9-82b5-eac164630fe9.png)

Look forward to your reply. Thanks.

**Comment by caozhangjie** [comment]
As described in the paper, exp(-H) takes values between 1/C and 1, where C is the number of classes. Namely, the maximal and minimal values of exp(-H) differ by C times, which may penalize too much for some examples. Hence, we use 1+exp(-H), the reciprocal of the sigmoid function, to shape the weights between (roughly) 1 and 2, which stabilizes training.

**Comment by shopeth** [comment]
I have a question on line 33 in here. coeff records the training progress. How does it change the entropy calculation?
Thank you for sharing the code.
